src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
I think you're missing a `` ` `` here.  I only noticed because I'm confused about your documentation test.

Maybe I'm missing something, but I think this won't work properly if one of the `Node`s being swapped is a direct descendant of the other `Node`.  In fact, I'm having a hard time wrapping my head around what _should_ happen in that scenario.

Also more generally, what is the expected behavior if one `Node` is in the sub-tree of the other `Node`?